core/sched.c

```diff
@@ -92,7 +92,7 @@ int __attribute__((used)) sched_run(void)
     thread_t *next_thread = container_of(sched_runqueues[nextrq].next->next, thread_t, rq_entry);
 
     DEBUG("sched_run: active thread: %" PRIkernel_pid ", next thread: %" PRIkernel_pid "\n",
-          (active_thread == NULL) ? KERNEL_PID_UNDEF : active_thread->pid,
+          (int16_t)((active_thread == NULL) ? KERNEL_PID_UNDEF : active_thread->pid),
```

Comment:
since the format is PRIkernel_pid, the cast should be kernel_pid_t

Reply:
yep, good point!